framework/DataObjects/XDataSet.py

```diff
@@ -206,13 +206,52 @@ def addVariable(self,varName,values,classify='meta'):
       self._metavars.append(varName)
     self._allvars.append(varName)
 
-  def asDataset(self):
+  def asDataset(self, outType=None):
+    """
+      Casts this dataObject as dictionary or an xr.Dataset depending on outType.
+      @ In, outType, str, type of output object (xr.Dataset or dictionary).
+      @ Out, xr.Dataset or dictionary.
+    """
+    if outType=='None' or outType=='xrDataset':
```

Comment:
Note that this will not check if `outType == None` since there's quotes around the None